### DISCUSSION TITLE
How to update shinny dashboard in shinnyapp.io

### DISCUSSION THREAD

**Discussion by DiekiRianMustapa** [teamalgoritma/community#316] [Q&A] [answered]
How to update shinny dashboard in shinnyapp.io without restart the shinnyapp.io ? i use google spreed sheet as data source, when the spreed sheet is updated the shinny dashboard should be restart with this button to update the dashboard with the new data...please help 

![image](https://user-images.githubusercontent.com/107328516/191475616-e5395f64-afaf-47de-9a78-b0d9b272ca30.png)

**Comment by NabiilahArdini** [comment, marked as answer]
Hello @DiekiRianMustapa , 
I've been exploring this topic too, but still stuck in configuration for partially protected sheet. 
Using googlesheet4 package will help you. It's an interface for Google Sheets to R, therefore you can also add it into your shiny. You need to first configure API from Google, embed the authentication script into global.R and read the sheet using function `read_sheet()`. For further use, you can use the usual R script. 
Hope this documentation helps https://googlesheets4.tidyverse.org/, and if I found any other resources, I'll share it too.

**Reply by DiekiRianMustapa** [reply]
yes, i already connect it with googlesheet4 and success, but my problem is the dashboard doesn't update automatically if the spreadsheet is changing/updated. it is really annoying if i use dynamic source  data (google spreadsheet)

**Reply by DiekiRianMustapa** [reply]
the dashboard can update if i restart the instance the file

**Reply by DiekiRianMustapa** [reply]
Hallo ?

**Reply by NabiilahArdini** [reply]
Hello, have you tried `invalidateLater()` function from shiny? This set how frequent your code needs to refresh. Adding it in the code for obtaining data might help you too: https://shiny.rstudio.com/reference/shiny/0.14/invalidateLater.html
